pkg/controller/operators/olm/operator.go

```diff
@@ -550,14 +567,66 @@ func (a *Operator) syncObject(obj interface{}) (syncError error) {
 		"self":      metaObj.GetSelfLink(),
 	})
 
-	// Requeue all owner CSVs
-	if ownerutil.IsOwnedByKind(metaObj, v1alpha1.ClusterServiceVersionKind) {
-		logger.Debug("requeueing owner csvs")
-		a.requeueOwnerCSVs(metaObj)
+	switch metaObj.(type) {
+	case *corev1.ServiceAccount:
+		if err := a.opClient.DeleteServiceAccount(metaObj.GetNamespace(), metaObj.GetName(), &metav1.DeleteOptions{}); err != nil {
+			logger.WithError(err).Warn("cannot delete service account")
+			syncError = err
+		}
+		logger.Debugf("Deleted apiservice %v due to no owning CSV", metaObj.GetName())
+	case *rbacv1.ClusterRole:
+		if err := a.opClient.DeleteClusterRole(metaObj.GetName(), &metav1.DeleteOptions{}); err != nil {
+			logger.WithError(err).Warn("cannot delete cluster role")
+			syncError = err
+		}
+		logger.Debugf("Deleted cluster role %v due to no owning CSV", metaObj.GetName())
+	case *rbacv1.ClusterRoleBinding:
+		if err := a.opClient.DeleteClusterRoleBinding(metaObj.GetName(), &metav1.DeleteOptions{}); err != nil {
+			logger.WithError(err).Warn("cannot delete cluster role binding")
+			syncError = err
+		}
+		logger.Debugf("Deleted cluster role binding %v due to no owning CSV", metaObj.GetName())
+	}
+
+	return
+}
+
+func (a *Operator) syncObject(obj interface{}) (syncError error) {
+	// Assert as metav1.Object
+	metaObj, ok := obj.(metav1.Object)
+	if !ok {
+		syncError = errors.New("object sync: casting to metav1.Object failed")
+		a.logger.Warn(syncError.Error())
+		return
 	}
+	logger := a.logger.WithFields(logrus.Fields{
+		"name":      metaObj.GetName(),
+		"namespace": metaObj.GetNamespace(),
+		"self":      metaObj.GetSelfLink(),
+	})
 
 	// Requeues objects that can't have ownerrefs (cluster -> namespace, cross-namespace)
 	if ownerutil.IsOwnedByKindLabel(metaObj, v1alpha1.ClusterServiceVersionKind) {
```

Comment:
There seems to be some duplicated work. I wonder if we can rearrange things to make it a little cleaner. Does it make sense to move some of this to the `requeueOwnerCSVs` method; particularly the existence check and GC enqueuing?

Reply:
Most of the code here was already present. The goal was to avoid requeueing CSVs in a deletion scenario. Maybe I can circle back to this later.